mllib/src/main/scala/org/apache/spark/ml/Transformer.scala

```diff
@@ -90,7 +90,7 @@ abstract class UnaryTransformer[IN, OUT, T <: UnaryTransformer[IN, OUT, T]]
    * account of the embedded param map. So the param values should be determined solely by the input
    * param map.
    */
-  protected def createTransformFunc: IN => OUT
+  protected val createTransformFunc: (T, IN) => OUT
```

Comment:
After re-thinking about this change, I think it is not a good approach. It works but hacky.